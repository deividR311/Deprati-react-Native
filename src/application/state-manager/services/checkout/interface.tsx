import { DeliveryOptionsResponse } from '../../../../infrastucture/apis/delivery-address';
import { Cart } from '../../../../infrastucture/apis/shopping-cart';
import { CheckoutSteps } from '../../../../presentation/navigation/checkout/interfaces';

export interface CheckoutStateHook {
  onContinueButtonTrigger?: CheckoutSteps;
  onPressContinueButton: (screenId: CheckoutSteps) => void;
}

export interface CheckoutState {
  screenId?: CheckoutSteps;
  previousScreen?: CheckoutSteps;
  stackSummary?: any[];
  showLoadingScreen?: boolean;
  onClickedContinueButton: boolean;
  cart: Partial<Cart>;
  totalPriceBeforeSelectDeliveryMode?: string;
  needRefreshCart?: boolean;
  deliveryOptions?: DeliveryOptionsResponse;
  isGiftCardCart: boolean;
  creditInfo?: {
    saveForFuturePay: boolean;
    maskedPhoneNumber: string;
    maskedNumberAccount: string;
    maskedEmail: string;
    directCreditCodeCustomer?: string;
    directCreditCodeAdditional?: string;
  };
  firstTimeSummary?: boolean;
}

export const INITIAL_STATE: CheckoutState = {
  screenId: undefined,
  previousScreen: undefined,
  stackSummary: undefined,
  onClickedContinueButton: false,
  showLoadingScreen: false,
  needRefreshCart: false,
  cart: {},
  deliveryOptions: undefined,
  totalPriceBeforeSelectDeliveryMode: undefined,
  isGiftCardCart: false,
  creditInfo: undefined,
  firstTimeSummary: undefined
};
