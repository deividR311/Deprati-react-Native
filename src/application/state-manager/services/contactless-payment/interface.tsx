import { ContactLessPaymentSteps } from '../../../../presentation/navigation/contactless-payment';

export enum BottomSheetType {
  IDENTITY_VALIDATION = 'IDENTITY_VALIDATION',
  SUCCESS_BUY = 'SUCCESS_BUY'
}

export interface ShowBottomSheetAction {
  bottomSheet: BottomSheetType;
  show: boolean;
  text?: string;
  ticketId?: string;
}

export interface ContactlessPaymentStateHook {
  TicketNumberContactLessPayment?: string;
  showIdentityValidation(show: boolean): void;
  identityValidationIsOpen: boolean;
  showSuccessBuy(show: boolean, text?: string, ticketId?: string): void;
  successBuyIsOpen: boolean;
  successBuyText?: string;
  successBuyTicketId?: string;
  doEnableConfirmPurchaseButton(isEnable: boolean, ticketId?: string): void;
  enableConfirmPurchaseButton: boolean;
}

export interface ContactlessPaymentState {
  tabbar: {
    disableNextButton: boolean;
    disablePreviousButton: boolean;
    currentTab: ContactLessPaymentSteps | '';
  };
  bottomSheetState: {
    identityValidationIsOpen: boolean;
    successBuyIsOpen: boolean;
    successBuyText: string;
    successBuyTicketId: string;
  };
  enableConfirmPurchaseButton: boolean;
}

export const INITIAL_STATE: ContactlessPaymentState = {
  tabbar: {
    currentTab: '',
    disableNextButton: false,
    disablePreviousButton: false
  },
  bottomSheetState: {
    identityValidationIsOpen: false,
    successBuyIsOpen: false,
    successBuyText: '',
    successBuyTicketId: ''
  },
  enableConfirmPurchaseButton: false
};
