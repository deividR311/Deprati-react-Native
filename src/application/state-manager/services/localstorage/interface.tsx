import { NotificationListRes } from '../../../../infrastucture/apis/notifications/notifications.interfaces';
import { MessagesAppInterface } from '../../../../presentation/screens/Splash/hooks/messages.interface';
import { INITIAL_STATE as ContactlessInitialState } from '../contactless-payment';
import {
  UserResponse,
  USER_DEFAULT_DATA
} from '../../../../infrastucture/apis/user/user.interface';

export type LocalStorageParams = Partial<LocalStorageType>;

export interface LocalStorageHook {
  localStorageData: LocalStorageType;
  save(params: LocalStorageParams): void;
  remove(keys: LocalStorageKey[]): void;
  readAndStorageInState(): void;
  fullClean(): void;
}

export enum LocalStorageKey {
  // BEHAVIOR
  WasStoreRead = 'WAS_STORE_READ',
  isAppOpenByFirstTime = 'IS_APP_OPEN_BY_FIRST_TIME',
  IpAddress = 'IP_ADDRESS',
  UID = 'DEVICE_ID',
  IsEnableConfirmPurchaseButtonInQRScreen = 'CONTACTLESS_PAYMENT@IS_ENABLE_CONFIRM_PURCHASE_BUTTON',
  TicketNumberContactLessPayment = 'CONTACTLESS_PAYMENT@TICKET_NUMBER',
  AnonymousCartGuid = 'ANONYMOUS_CART_GUID',

  // USER DATA
  User = 'USER_DATA',
  UserEmail = 'USER_EMAIL',
  UserLastEntry = 'USER_LAST_ENTRY',

  // ACCOUNT DATA
  IsAccountAuthenticated = 'IS_ACCOUNT_AUTHENTICATED',
  AccountNumber = 'ACCOUNT_NUMBER',
  AccountDisplayNumber = 'ACCOUNT_DISPLAY_NUMBER',
  AccountAdditionalNumber = 'ACCOUNT_ADDITIONAL_NUMBER',
  AccountPhoneNumber = 'ACCOUNT_DISPLAY_PHONE_NUMBER',
  LastCustomerId = 'LAST_CUSTOMER_ID',

  //Notifications
  Notifications = 'NOTIFICATIONS',

  //Notifications Push FCM
  TokenFCM = 'TOKENFCM',

  // SECURITY
  Token = 'TOKEN',
  RefreshToken = 'REFRESH_TOKEN',
  ExpiresIn = 'EXPERES_IN',
  ExpiresAt = 'EXPERES_AT',
  GetTokenDate = 'GET_TOKEN_DATE',
  IsLogin = 'IS_LOGIN',
  IsSucessFirstLogin = 'IS_SUCCESS_FIRST_LOGIN',

  BiometricPublicKey = 'publicKey-generate',
  MessagesApp = 'MESSAGES_APP',
  LocalStorageKey = 'LocalStorageKey',
  Whatsapp = 'WHATSAPP'
}

export interface LocalStorageType {
  [LocalStorageKey.WasStoreRead]: boolean;
  [LocalStorageKey.isAppOpenByFirstTime]: boolean;
  [LocalStorageKey.UserLastEntry]: string;
  [LocalStorageKey.UserEmail]: string;
  [LocalStorageKey.AccountNumber]: string;
  [LocalStorageKey.AccountDisplayNumber]: string;
  [LocalStorageKey.AccountAdditionalNumber]: string;
  [LocalStorageKey.AccountPhoneNumber]: string;
  [LocalStorageKey.LastCustomerId]: string;
  [LocalStorageKey.IsEnableConfirmPurchaseButtonInQRScreen]: boolean;
  [LocalStorageKey.TicketNumberContactLessPayment]?: string;
  [LocalStorageKey.AnonymousCartGuid]?: string;
  [LocalStorageKey.User]: UserResponse;
  [LocalStorageKey.Token]: string;
  [LocalStorageKey.RefreshToken]: string;
  [LocalStorageKey.ExpiresIn]: number;
  [LocalStorageKey.ExpiresAt]: string;
  [LocalStorageKey.GetTokenDate]: string;
  [LocalStorageKey.IsLogin]: boolean;
  [LocalStorageKey.IsAccountAuthenticated]: boolean;
  [LocalStorageKey.IpAddress]: string;
  [LocalStorageKey.BiometricPublicKey]: string;
  [LocalStorageKey.UID]: string;
  [LocalStorageKey.IsSucessFirstLogin]: boolean;
  [LocalStorageKey.Notifications]: {
    unread: number;
    notifications: NotificationListRes[];
  };
  [LocalStorageKey.MessagesApp]: MessagesAppInterface;
  [LocalStorageKey.TokenFCM]: string;
  [LocalStorageKey.Whatsapp]: {
    message: string;
    phone: string;
  };
}

export const InitialLocalStorage: LocalStorageType = {
  [LocalStorageKey.WasStoreRead]: false,
  [LocalStorageKey.isAppOpenByFirstTime]: true,
  [LocalStorageKey.AccountNumber]: '',
  [LocalStorageKey.AccountDisplayNumber]: '',
  [LocalStorageKey.AccountAdditionalNumber]: '',
  [LocalStorageKey.AccountPhoneNumber]: '',
  [LocalStorageKey.LastCustomerId]: '',
  [LocalStorageKey.User]: USER_DEFAULT_DATA,
  [LocalStorageKey.UserEmail]: '',
  [LocalStorageKey.UserLastEntry]: '',
  [LocalStorageKey.IsEnableConfirmPurchaseButtonInQRScreen]:
    ContactlessInitialState.enableConfirmPurchaseButton,
  [LocalStorageKey.TicketNumberContactLessPayment]: '',
  [LocalStorageKey.AnonymousCartGuid]: '',
  [LocalStorageKey.Token]: '',
  [LocalStorageKey.RefreshToken]: '',
  [LocalStorageKey.ExpiresIn]: 0,
  [LocalStorageKey.ExpiresAt]: '',
  [LocalStorageKey.GetTokenDate]: '',
  [LocalStorageKey.IsLogin]: false,
  [LocalStorageKey.IsAccountAuthenticated]: false,
  [LocalStorageKey.IsAccountAuthenticated]: false,
  [LocalStorageKey.IpAddress]: '0.0.0.0',
  [LocalStorageKey.BiometricPublicKey]: '',
  [LocalStorageKey.UID]: '',
  [LocalStorageKey.IsSucessFirstLogin]: false,
  [LocalStorageKey.Notifications]: {
    unread: 0,
    notifications: []
  },
  [LocalStorageKey.MessagesApp]: {},
  [LocalStorageKey.TokenFCM]: '',
  [LocalStorageKey.Whatsapp]: {
    phone: '',
    message: ''
  }
};
