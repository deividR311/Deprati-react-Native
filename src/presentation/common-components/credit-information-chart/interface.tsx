export interface CreditInformationChartProps {
  ownerFullname?: string;
  cardNumber?: string;
  affiliateDate?: string;
  totalAmount?: number;
  amountSpent?: number;
  availableAmount?: number;
  numeroTarjetaAdicionalDisplay?: string;
  showAffiliateDate?: boolean;
  showDisplayCard?: boolean;
  displayCartType?: 'small' | 'large';
  invertValues?: boolean;
  constentChart: any;
  wrapperChartInCard?: boolean;
}

export interface DonutChartProps {
  indicatorValue?: number;
  maxIndicatorValue?: number;
  color?: string;
  dialColor?: string;
  radius?: number;
  strokeWidth?: number;
  duration?: number;
  delay?: number;
  invertValues?: boolean;
}
