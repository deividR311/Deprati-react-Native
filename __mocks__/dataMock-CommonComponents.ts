export const DataMock_CustomProps = {
  url: 'url',
  style: {},
  content: 'qwerty',
  uid: 'uid',
  typeCode: 'qwerty',
  modifiedTime: 'qwerty',
  name: 'qwerty',
  container: true,
  rotatingImagesComponentsList: 'DepratiBrandsRotatingImages',
  childrenComponentsList: ['DepratiBrandsRotatingImages'],
  childrenComponents: {
    DepratiBrandsRotatingImages: {
      uid: 'DepratiBrandsRotatingImages',
      typeCode: 'RotatingImagesComponent',
      modifiedTime: '2023-04-28T12:59:58+0000',
      name: 'DePrati Dior Mobile Carousel',
      container: 'false',
      resolutionEnum: 'MOBILE',
      effect: 'FADE',
      banners: 'DiorBannerComponent2 BannerCarouselDiorVideo_1',
      timeout: '6',
      childrenComponentsList: [
        'DiorBannerComponent2',
        'BannerCarouselDiorVideo_1'
      ]
    }
  }
};

export const DataMock_BillingAddressComponent = {
  style: {},
  onSelected: () => {},
  emptyStyle: {},
  showTitle: true,
  enableHandleEnableButton: true,
  onIsEnable: () => {},
  paymentAddress: {
    defaultAddress: false,
    email: 'string',
    firstName: 'string',
    formattedAddress: 'string',
    id: 'string',
    line1: 'string',
    line2: 'string',
    phone: 'string',
    phonePreffix: 'string',
    shippingAddress: false,
    visibleInAddressBook: false
  },
  paymentCredit: {
    showAlerAddressPayment: false,
    setShowAlerAddressPayment: () => {}
  },
  styleSecundaryButton: false
};

export const DataMock_useBillingAddress = {
  addressSelected: {
    defaultAddress: false,
    firstName: 'firstName',
    id: 'qwertystring',
    line1: 'qwerty',
    line2: 'qwerty',
    phone: 'qwerty',
    phonePreffix: 'qwerty'
  },
  isLoadingByDeleting: false,
  addressPaymentList: {
    default: null,
    others: [],
    existsAddress: false,
    loading: false
  },
  isLoading: false,
  buttonChangeData: false,
  handleChangeData: jest.fn(),
  changeAddress: {
    handleSelected: jest.fn(),
    addressSelected: undefined
  },
  handleDelete: jest.fn(),
  handleAddEdit: jest.fn(),
  handleBillingAddressList: () => {
    return [
      {
        defaultAddress: false,
        email: 'string',
        firstName: 'string',
        formattedAddress: 'string',
        id: 'string',
        line1: 'string',
        line2: 'string',
        phone: 'string',
        phonePreffix: 'string',
        shippingAddress: false,
        visibleInAddressBook: false
      }
    ];
  }
};

export const DataMock_IncreaseCreditLimit = {
  rectangularSize: 232,
  value: 1000,
  maxValue: 2000,
  onChangeCreditLimit: jest.fn()
};

export const DataMock_Charts = {
  constentChart: {
    availableAmount: 100,
    amountSpent: 20,
    totalAmount: 200
  },
  ownerFullname: 'Custom customer customization',
  invertValues: true,
  showDisplayCard: true
};
