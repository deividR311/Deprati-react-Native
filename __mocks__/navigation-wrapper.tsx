import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
