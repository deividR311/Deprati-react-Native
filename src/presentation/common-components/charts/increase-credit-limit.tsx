import React, { FC } from 'react';
import { StyleSheet, Text, View, Animated } from 'react-native';
import { Svg, Path } from 'react-native-svg';
import { COLORS, FontStyles } from '../../../application/common';
import { currencyFormatter } from '../../../application/utils/currency';
import AnimatedCursor from './arc/chart-cursor';
import {
  fromValueToPercentage,
  fromPercentageToRad,
  fromRadToProgress,
  getLargeArcValue
} from './arc/chart-math-utils';
import { fontFamilyOS, fontWeightOS } from '../../../application/utils';
import { t } from 'i18next';

const AnimatedArc = Animated.createAnimatedComponent(Path);

export const ChartIncreaseCreditLimit: FC<ChartIncreaseCreditLimitProps> = ({
  rectangularSize,
  value,
  maxValue,
  holdInValue = false,
  step = 1,
  onChangeCreditLimit
}) => {
  const STROKE_WIDTH = 16;
  const { PI, cos, sin } = Math;
  const r = (rectangularSize - STROKE_WIDTH) / 2;
  const cx = rectangularSize / 2;
  const cy = rectangularSize / 2;
  const A = PI + PI * 0.4;
  const circumference = r * A;

  const [progress, setProgress] = React.useState(value);
  const [progressToShow, setProgressToShow] = React.useState(value);

  // Calculamos los radianes que va a ocupar arco con su correspondiente desplazamiento

  const indicatorViewPercentage = fromValueToPercentage(progress, maxValue);
  const indicatorViewRadians = fromPercentageToRad(indicatorViewPercentage);
  const indicatiorViewConvertion = fromRadToProgress(indicatorViewRadians, 0);

  const angle = PI - indicatorViewRadians + Math.PI * 0.2;

  // Container Available Value props
  const availableValueProps = {
    startAngle: PI + PI * 0.2,
    endAngle: 2 * PI - PI * 0.2,
    get x1() {
      return cx - r * cos(this.startAngle);
    },
    get y1() {
      return -r * sin(this.startAngle) + cy;
    },
    get x2() {
      return cx - r * cos(this.endAngle);
    },
    get y2() {
      return -r * sin(this.endAngle) + cy;
    }
  };

  // Adelantados arc props
  const indicatorViewProps = {
    startAngle: indicatiorViewConvertion,
    endAngle: 2 * PI - PI * 0.2,
    get x1() {
      return cx - r * cos(this.startAngle);
    },
    get y1() {
      return -r * sin(this.startAngle) + cy;
    },
    get x2() {
      return cx - r * cos(this.endAngle);
    },
    get y2() {
      return -r * sin(this.endAngle) + cy;
    }
  };

  // Large arc (>180)? 0 = NO | 1 = YES
  // Si beta es negativo, le sumamos 2PI a el valor de advance en radianes para conseguir la equivalencia positiva
  const laAd = getLargeArcValue(
    indicatorViewProps.endAngle,
    angle < 0 ? 2 * PI + indicatorViewRadians : indicatiorViewConvertion,
    true,
    angle
  );

  const pathAvailableValue = `M ${availableValueProps.x1} ${availableValueProps.y1} A ${r} ${r} 0 1 0 ${availableValueProps.x2} ${availableValueProps.y2}`;
  const pathIndicatorView = `M ${indicatorViewProps.x1} ${indicatorViewProps.y1} A ${r} ${r} 0 ${laAd} 0 ${indicatorViewProps.x2} ${indicatorViewProps.y2}`;

  const _onChangeValue = (newValue: number) => {
    let _newValue =
      value >= newValue && holdInValue
        ? value
        : maxValue <= newValue
        ? maxValue
        : newValue;
    setProgress(_newValue);
    setProgressToShow(_newValue - (_newValue % step));
    onChangeCreditLimit?.(_newValue - (_newValue % step));
  };

  return (
    <View style={[styles.container]}>
      {/* Labels */}
      <View style={[styles.textContainer]}>
        <Text
          style={[FontStyles.Body_2, FontStyles.MutedColor, FontStyles.Center]}>
          {t('desiredNewSpace')}
        </Text>
        <View style={styles.space} />
        <Text
          style={[
            FontStyles.PrimaryColor,
            styles.amountText,
            FontStyles.Center
          ]}>
          {currencyFormatter(progressToShow)}
        </Text>
      </View>
      {/* ARCs */}
      <Animated.View
        style={[
          styles.progressView,
          {
            width: rectangularSize,
            height: rectangularSize
          }
        ]}>
        <Svg width={rectangularSize} height={rectangularSize}>
          <AnimatedArc
            stroke={COLORS.GRAYBRAND}
            fill="none"
            strokeLinecap="round"
            strokeDasharray={`${circumference}, ${circumference}`}
            strokeWidth={STROKE_WIDTH}
            d={pathAvailableValue}
          />
          <AnimatedArc
            stroke={COLORS.BRAND}
            fill="none"
            strokeLinecap="round"
            strokeDasharray={`${circumference}, ${circumference}`}
            strokeWidth={STROKE_WIDTH}
            d={pathIndicatorView}
          />
        </Svg>
      </Animated.View>
      {/* Cursors */}
      <AnimatedCursor
        minValue={holdInValue ? value : 0}
        radius={r}
        beta={PI + PI * 0.2}
        alpha={angle}
        maxValue={maxValue}
        area={rectangularSize}
        onChangeValue={_onChangeValue}
      />
    </View>
  );
};

interface ChartIncreaseCreditLimitProps {
  rectangularSize: number;
  holdInValue?: boolean;
  value: number;
  maxValue: number;
  step?: number;
  onChangeCreditLimit?: (value: number) => void;
}

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center'
  },
  textContainer: {
    marginTop: -20
  },
  amountText: {
    fontSize: 30,
    fontWeight: fontWeightOS('500'),
    fontFamily: fontFamilyOS(),
    lineHeight: 40,
    letterSpacing: 0.25,
    textAlign: 'left'
  },
  space: {
    height: 8
  },
  progressView: {
    position: 'absolute'
  }
});
