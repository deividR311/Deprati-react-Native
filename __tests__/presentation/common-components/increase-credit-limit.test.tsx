import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../__mocks__/navigation-wrapper';
import { ChartIncreaseCreditLimit } from '../../../src/presentation/common-components/charts/increase-credit-limit';
import { DataMock_IncreaseCreditLimit } from '../../../__mocks__/dataMock-CommonComponents';
import { IRender } from '../../../__mocks__/IRender.interface';

let screenTest: IRender;
describe('ChartIncreaseCreditLimit', () => {
  it('renders correctly ChartIncreaseCreditLimit in Snapshot', () => {
    screenTest = render(
      <NavigationWrapper
        screen={() => (
          <ChartIncreaseCreditLimit {...DataMock_IncreaseCreditLimit} />
        )}
      />
    );
    expect(screenTest.toJSON()).toMatchSnapshot();
  });

  it('renders correctly ChartIncreaseCreditLimit', () => {
    screenTest = render(
      <NavigationWrapper
        screen={() => (
          <ChartIncreaseCreditLimit
            {...DataMock_IncreaseCreditLimit}
            holdInValue
          />
        )}
      />
    );
    expect(screenTest.toJSON()).toMatchSnapshot();
  });
});
