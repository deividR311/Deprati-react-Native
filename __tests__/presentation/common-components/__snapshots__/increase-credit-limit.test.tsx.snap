// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartIncreaseCreditLimit renders correctly ChartIncreaseCreditLimit 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginTop": -20,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "letterSpacing": 0.25,
                    "lineHeight": 20,
                  },
                  Object {
                    "color": "#26323880",
                  },
                  Object {
                    "textAlign": "center",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "height": 8,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#E31A2C",
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 30,
                    "fontWeight": undefined,
                    "letterSpacing": 0.25,
                    "lineHeight": 40,
                    "textAlign": "left",
                  },
                  Object {
                    "textAlign": "center",
                  },
                ]
              }
            >
              $1.000,00
            </Text>
          </View>
          <View
            collapsable={false}
            style={
              Object {
                "height": 232,
                "position": "absolute",
                "width": 232,
              }
            }
          >
            <RNSVGSvgView
              bbHeight={232}
              bbWidth={232}
              focusable={false}
              height={232}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 232,
                    "width": 232,
                  },
                ]
              }
              width={232}
            >
              <RNSVGGroup>
                <RNSVGPath
                  d="M 203.37383539249433 179.48080724758708 A 108 108 0 1 0 28.626164607505686 179.4808072475871"
                  fill={null}
                  propList={
                    Array [
                      "fill",
                      "stroke",
                      "strokeWidth",
                      "strokeDasharray",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4293322470}
                  strokeDasharray={
                    Array [
                      "475.00880922277673",
                      "475.00880922277673",
                    ]
                  }
                  strokeLinecap={1}
                  strokeWidth={16}
                />
                <RNSVGPath
                  d="M 116 8 A 108 108 0 0 0 28.626164607505686 179.4808072475871"
                  fill={null}
                  propList={
                    Array [
                      "fill",
                      "stroke",
                      "strokeWidth",
                      "strokeDasharray",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4293073452}
                  strokeDasharray={
                    Array [
                      "475.00880922277673",
                      "475.00880922277673",
                    ]
                  }
                  strokeLinecap={1}
                  strokeWidth={16}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              Array [
                Object {
                  "position": "absolute",
                },
                Object {
                  "height": 232,
                  "width": 232,
                },
              ]
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              handlerTag={2}
              handlerType="PanGestureHandler"
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "backgroundColor": "#E31A2C",
                  "borderColor": "#E6E6E6",
                  "borderRadius": 19,
                  "borderWidth": 5,
                  "bottom": 0,
                  "elevation": 4,
                  "height": 39,
                  "left": 0,
                  "marginLeft": -11,
                  "marginTop": -11,
                  "position": "absolute",
                  "right": 0,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "translateX": 108,
                    },
                  ],
                  "width": 39,
                }
              }
            />
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;

exports[`ChartIncreaseCreditLimit renders correctly ChartIncreaseCreditLimit in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginTop": -20,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "letterSpacing": 0.25,
                    "lineHeight": 20,
                  },
                  Object {
                    "color": "#26323880",
                  },
                  Object {
                    "textAlign": "center",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "height": 8,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#E31A2C",
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 30,
                    "fontWeight": undefined,
                    "letterSpacing": 0.25,
                    "lineHeight": 40,
                    "textAlign": "left",
                  },
                  Object {
                    "textAlign": "center",
                  },
                ]
              }
            >
              $1.000,00
            </Text>
          </View>
          <View
            collapsable={false}
            style={
              Object {
                "height": 232,
                "position": "absolute",
                "width": 232,
              }
            }
          >
            <RNSVGSvgView
              bbHeight={232}
              bbWidth={232}
              focusable={false}
              height={232}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 232,
                    "width": 232,
                  },
                ]
              }
              width={232}
            >
              <RNSVGGroup>
                <RNSVGPath
                  d="M 203.37383539249433 179.48080724758708 A 108 108 0 1 0 28.626164607505686 179.4808072475871"
                  fill={null}
                  propList={
                    Array [
                      "fill",
                      "stroke",
                      "strokeWidth",
                      "strokeDasharray",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4293322470}
                  strokeDasharray={
                    Array [
                      "475.00880922277673",
                      "475.00880922277673",
                    ]
                  }
                  strokeLinecap={1}
                  strokeWidth={16}
                />
                <RNSVGPath
                  d="M 116 8 A 108 108 0 0 0 28.626164607505686 179.4808072475871"
                  fill={null}
                  propList={
                    Array [
                      "fill",
                      "stroke",
                      "strokeWidth",
                      "strokeDasharray",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4293073452}
                  strokeDasharray={
                    Array [
                      "475.00880922277673",
                      "475.00880922277673",
                    ]
                  }
                  strokeLinecap={1}
                  strokeWidth={16}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              Array [
                Object {
                  "position": "absolute",
                },
                Object {
                  "height": 232,
                  "width": 232,
                },
              ]
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              handlerTag={1}
              handlerType="PanGestureHandler"
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "backgroundColor": "#E31A2C",
                  "borderColor": "#E6E6E6",
                  "borderRadius": 19,
                  "borderWidth": 5,
                  "bottom": 0,
                  "elevation": 4,
                  "height": 39,
                  "left": 0,
                  "marginLeft": -11,
                  "marginTop": -11,
                  "position": "absolute",
                  "right": 0,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "translateX": 108,
                    },
                  ],
                  "width": 39,
                }
              }
            />
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
